roofit/roofitcore/src/RooProdPdf.cxx

```diff
@@ -712,15 +711,14 @@ void RooProdPdf::factorizeProduct(const RooArgSet& normSet, const RooArgSet& int
   }
 
   // Loop over list of terms again to determine 'imported' observables
-  RooArgSet *normDeps, *allDeps, *intNoNormDeps;
+  int i = 0;
+  RooArgSet *normDeps, *allDeps;
   for (RooFIter lIter = termList.fwdIterator(),
       ldIter = normList.fwdIterator(),
-      laIter = depAllList.fwdIterator(),
-      innIter = depIntNoNormList.fwdIterator();
+      laIter = depAllList.fwdIterator();
       (normDeps = (RooArgSet*) ldIter.next(),
        allDeps = (RooArgSet*) laIter.next(),
-       intNoNormDeps = (RooArgSet*) innIter.next(),
-       term=(RooArgSet*)lIter.next()); ) {
+       term=(RooArgSet*)lIter.next()); ++i) {
     // Make list of wholly imported dependents
     RooArgSet impDeps(*allDeps);
     impDeps.remove(*normDeps, kTRUE, kTRUE);
```

Comment:
Away with those pointers that are declared outside of for/while loops with weird iterators.

RooFit has two kinds of loops: range-based for for a single element and for multi-loops e.g.
```c++
for (unsigned int i = ...) {
  auto ld = static_cast<...>(normList[i]);
  auto la = static_cast<...>(depAllList[i]);
```

or if you like
```c++
for (auto a = as.begin(), auto b = b.begin(); a < as.end() && b < bs.end(); ++a; ++b) { 
```